specification/network/resource-manager/Microsoft.Network/stable/2024-10-01/networkVirtualAppliance.json

```diff
@@ -1764,6 +1783,88 @@
         }
       },
       "description": "Specifies input parameters required to retrieve the boot diagnostic logs for an NVA instance"
+    },
+    "NvaInterfaceConfigurations": {
+      "type": "array",
+      "readOnly": false,
+      "description": "The NVA in VNet interface configurations details",
+      "minItems": 2,
+      "maxItems": 3,
+      "items": {
+        "$ref": "#/definitions/NvaInterfaceConfigurationsProperties"
+      }
+    },
+    "NvaInterfaceConfigurationsProperties": {
+      "type": "object",
+      "properties": {
+        "subnet": {
+          "readOnly": false,
+          "$ref": "#/definitions/NvaInVnetSubnetReferenceProperties",
+          "description": "A subnet resource id where the NIC will be deployed. Each subnet resource uri should be unique.",
+          "x-ms-identifiers": []
+        },
+        "type": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": [
+              "PrivateNic",
+              "PublicNic",
+              "AdditionalPrivateNic",
+              "AdditionalPublicNic"
+            ],
+            "x-ms-enum": {
+              "name": "NvaNicType",
+              "modelAsString": true,
+              "values": [
+                {
+                  "value": "PrivateNic",
+                  "description": "The private NIC type"
+                },
+                {
+                  "value": "PublicNic",
+                  "description": "The public NIC type"
+                },
+                {
+                  "value": "AdditionalPrivateNic",
+                  "description": "An additional private NIC type"
+                },
+                {
+                  "value": "AdditionalPublicNic",
+                  "description": "An additional public NIC type"
+                }
+              ]
+            }
+          },
+          "description": "Specifies the NIC types for the NVA interface configuration. Allowed values: PrivateNic, PublicNic, AdditionalPrivateNic, AdditionalPublicNic. Only the combination of PrivateNic and PublicNic is currently supported."
+        },
+        "name": {
+          "type": "string",
+          "maxLength": 70,
+          "description": "Specifies the name of the interface. Maximum length is 70 characters."
+        }
+      },
+      "description": "Specifies input parameters required NVA in VNet interface configuration."
+    },
+    "NvaInVnetSubnetReferenceProperties": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

[ARMBlockingComment] Rename to something different from top-level properties (name, type, id, systemData, etc.). Recommend something the includes resourceId in the name, like resourceId or subnetResourceId, or something you like better.